ignite/handlers/time_limit.py

```diff
@@ -0,0 +1,46 @@
+import logging
+import time
+from typing import Optional
+
+from ignite.engine import Engine
+
+__all__ = ["TimeLimit"]
+
+
+class TimeLimit:
+    """TimeLimit handler can be used to control training time for computing environments where session time is limited.
+    Timer starts when handler is created and not training started.
+    This handler gracefully terminates the training if time passed in the training exceeds a limit.
+
+    Args:
+        limit_sec (int, optional): Maximum time before training terminates (in seconds). Defaults to 28800.
+
+    Examples:
+
+        .. code-block:: python
+
+            from ignite.engine import Events
+            from ignite.handlers import TimeLimit
+
+            handler = TimeLimit() # 8 hours of training
+            trainer.add_event_handler(Events.ITERATION_COMPLETED, handler)
+
+    .. versionadded:: 0.4.3
+    """
+
+    def __init__(self, limit_sec: Optional[int] = 28800):
+
+        if not isinstance(limit_sec, int):
+            raise TypeError("Argument limit_sec should be an integer.")
+        if limit_sec <= 0:
+            raise ValueError("Argument limit_sec should be a positive integer.")
+
+        self.limit_sec = limit_sec
+        self.start_time = time.time()
+        self.logger = logging.getLogger(__name__ + "." + self.__class__.__name__)
```

Comment:
Please use `setup_logger` from `ignite.utils`

```suggestion
        self.logger = setup_logger(__name__ + "." + self.__class__.__name__)
```

Reply:
@sdesrozis I think it is OK for now as we do the same thing for other handlers. If we have to change that, let's do in another PR.

Reply:
Ok !!